Library/Homebrew/utils/curl.rb

```diff
@@ -40,3 +40,30 @@ def curl_output(*args)
     [stdout.read, stderr.read, wait_thread.value]
   end
 end
+
+def url_status_code(url, range: false, user_agent: :default)
```

Comment:
Would rather leave this in `audit` until it's used elsewhere, I think. It also makes it easier to see what the changes have been (if any).